### DISCUSSION TITLE
Using GitHub PAT from the git credential store. Error: Failed to install 'unknown package' from GitHub:   HTTP error 401.   Bad credentials

### DISCUSSION THREAD

**Discussion by Kaikash777** [community/community#140956] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Hello, 

I was wondering if someone can please give me advice as I'm a complete novice. 

I'm trying to download a package called rmaxent from Github and apparently I have bad credentials. I really don't understand how to solve this issue, I don't understand what 'PAT' means, I don't think I've ever got a token,  and I have put my attempts to find a solution in this post below. 

Many thanks in advance if anyone can help me. 

Attempt 1:

#install the package maxent
library(devtools)
install.packages("remotes")
remotes::install_github("johnbaums/rmaxent")

Error Message

Using GitHub PAT from the git credential store.
Error: Failed to install 'unknown package' from GitHub:
  HTTP error 401.
  Bad credentials

  Rate limit remaining: 59/60
  Rate limit reset at: 2024-10-08 10:42:59 UTC

Attempts at correcting the bad credentials with Github

I then tried gitcreds::gitcreds_set() 

I was given a selection:

1: Abort update with error, and keep the existing credentials; 
2: Replace these credentials; 
3: See the password / token. 

I chose selection 2, and this was the outcome

-> Adding new credentials...
-> Removing credentials from cache...
-> Done.

I then tried this:

gitcreds::gitcreds_delete()

-> What would you like to do? 

1: Abort update with error, and keep the existing credentials
2: Delete these credentials
3: See the password / token

I tried looking at selection 3 and I wasn't able to see the token or the password, and I was asked to make a selection again, so I chose selection 2

-> Removing current credentials...
-> Removing credentials from cache...
-> Done.

It doesn't matter what section I choose, this process doesn't do anything and the selection menu asks me to choose a selection again.

**Comment by luddthree** [comment]
you need to generate a pat token, this might work:

Go to [Github/settings/tokens](https://github.com/settings/tokens)
Click "Generate new token." (classic probably the best)
Select the appropriate scopes for your use case. For installing packages, the following scopes are usually enough:

> repo 
> read:packages

Click "Generate token."
Copy the token, as you'll need it soon.

Set up the PAT in R:

Open R and run the following commands to set the new token:
`gitcreds::gitcreds_set()`

This will prompt you to enter a new credential. Paste your newly generated PAT when prompted.

Try Installing the Package Again: Now that your PAT is set, try running:
`remotes::install_github("johnbaums/rmaxent")`

This should work without the "Bad credentials" error if the PAT is correctly set.


If your repository is private, ensure you have the right access permissions.
If your PAT is not set properly, you can delete and set it again using:

```
gitcreds::gitcreds_delete()  # To remove any old tokens
gitcreds::gitcreds_set()     # To set the new PAT
```

**Reply by 10-Out-Of-10** [reply]
You dah best. Worked perfectly. Thanks!

**Reply by stellameku** [reply]
i was encountering the same problem, but  i still got an error reply, do u kno whats wrong with it :( i followed all ur instructions: Using GitHub PAT from the git credential store.
Downloading GitHub repo soumyaray/compstatslib@HEAD
Error: Failed to install 'compstatslib' from GitHub:
  there is no package called ‘processx’

**Reply by daytonamelia** [reply]
> i was encountering the same problem, but i still got an error reply, do u kno whats wrong with it :( i followed all ur instructions: Using GitHub PAT from the git credential store. Downloading GitHub repo soumyaray/compstatslib@HEAD Error: Failed to install 'compstatslib' from GitHub: there is no package called ‘processx’

Seems like you need to try installing 'processx' before installing 'compstatslib.' Solution above worked for me!

**Reply by ER-PL** [reply]
The best reply I have seen.

**Reply by Dia-xuan** [reply]
Saved me! Thank you
